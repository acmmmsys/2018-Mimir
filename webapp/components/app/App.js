import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import PageRoot from 'pages/PageRoot';
import ReportPage from 'pages/ReportPage';
import ImageAnalysisPage from 'pages/ImageAnalysisPage';

import Sidebar from 'components/sidebar/Sidebar';

import 'style/app/App.scss';

const App = () =>
    <div className="app" >
        <Sidebar />
        <PageRoot>
            <Switch>
                <Route exact path="/" component={ImageAnalysisPage} />
                <Route path="/image" component={ImageAnalysisPage} />
                <Route path="/report" component={ReportPage} />
            </Switch>
        </PageRoot>
    </div>

export default App;