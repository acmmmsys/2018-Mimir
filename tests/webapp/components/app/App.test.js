import React from 'react';
import App from '../../../../webapp/components/app/App'
import renderer from 'react-test-renderer';

test('App does render', () => {
  const wrapper = shallow(
      <App></App>
  );
  expect(wrapper).toMatchSnapshot();
});
