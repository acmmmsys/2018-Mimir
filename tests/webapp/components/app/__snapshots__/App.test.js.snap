// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App does render 1`] = `
<div
  className="app"
>
  <Sidebar />
  <PageRoot>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/image"
      />
      <Route
        component={[Function]}
        path="/report"
      />
    </Switch>
  </PageRoot>
</div>
`;
